.github/workflows/release.yml

```diff
@@ -0,0 +1,97 @@
+name: Build and release
+on:
+    push:
+      tags:
+        - 'v[0-9]+\.[0-9]+\.[0-9]+'
+
+jobs:
+    build:
+        name: build release binaries on ${{ matrix.os }}
+        strategy:
+            fail-fast: false
+            matrix:
+                os: ["ubuntu-latest"]
+        runs-on: ${{ matrix.os }}
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v4
+
+            - name: Install rust
+              id: rust
+              uses: dtolnay/rust-toolchain@stable
+
+            - name: Install cross
+              run: |
+                cargo install cross --git https://github.com/cross-rs/cross
+
+            - name: Build unix
+              id: unix_build
+              run: |
+                cross build --release --locked --target x86_64-unknown-linux-gnu
+                cross build --release --locked --target armv7-unknown-linux-gnueabihf
+                cross build --release --locked --target aarch64-unknown-linux-gnu
```

Comment:
Can we add macos & windows as well?, if possible.

Reply:
I tried with macos but gave up sorry.

Reply:
I'm happy to take a look at this. How were you testing it?